### PR TITLE
Add boards list mode

### PR DESCRIPTION
**Pull request by dantecatalfamo**
Adds a command that displays a list of all boards and their details. Allows users to easily add new boards to their projects or initialize new projects more easily.

image

### CODE REVIEW

File: platformio-mode.el [hunk 1]
```diff
@@ -96,6 +111,78 @@
...
                             runcmd)))
 
+
+;;; Board list functions
+(defun platformio--add-board (board)
+  "Add a BOARD to a PlatformIO porject."
+  (platformio--exec (string-join (list "init --ide emacs --board " board))))
+
+(defun platformio--build-board-table (_autoignore _noconfirm)
+  "Return a list of all boards supported by PlatforIO."
```
Comment (ZachMassia) [inline platformio-mode.el:121]:
```suggestion
  "Return a list of all boards supported by PlatformIO."
```

File: platformio-mode.el [hunk 2]
```diff
@@ -96,6 +111,78 @@
                               (platformio--silent-arg))
                             runcmd)))
 
+
+;;; Board list functions
+(defun platformio--add-board (board)
+  "Add a BOARD to a PlatformIO porject."
```
Comment (ZachMassia) [inline platformio-mode.el:117]:
```suggestion
  "Add a BOARD to a PlatformIO project."
```

**Comment by dantecatalfamo** [comment]
Thanks for catching those typos 😅

**Comment by ZachMassia** [comment]
No worries.

Thanks for all the contributions! I'm curious what brought about all this recent activity lol.

Little slow getting to this, been a crazy week at work. Looks good to go!

**Pull request merged**